src/transactions/TransactionFrame.cpp

```diff
@@ -1667,14 +1689,17 @@ TransactionFrame::applyExpirationBumps(Application& app, AbstractLedgerTxn& ltx)
         }
     }
 
+    bool isBumpOp = mOperations.front()->getOperation().body.type() ==
+                    BUMP_FOOTPRINT_EXPIRATION;
     // TODO: Write expiration extension entries instead of witing whole entry
     for (auto const& key : resources.footprint.readOnly)
     {
         auto lte = ltxTx.load(key);
         if (lte && isSorobanDataEntry(lte.current().data))
         {
             // Must enforce minimum expirations on write
-            bump(lte.current(), autoBumpEnabled(lte.current()), true);
+            bump(lte.current(), autoBumpEnabled(lte.current()) && !isBumpOp,
```

Comment:
I'm not a fan of breaking the op abstraction even further beyond classic vs soroban. Probably not for this PR, but we should try to make this less messy somehow. Maybe footprint really should be per-operation and then autobump would only happen in invoke host fn op...

Reply:
Yeah this isn't great. Instead of messing with the footprint (I think it's too late for this) it might make sense to move the autobump code to invoke host fn op. The other soroban ops (bumpFootprint and restoreFootprint) shouldn't use autobump.